Import project

zoltanszocs opened the pull request:
<!--
Thank you for your pull request. Please provide a description above and
review the checklist below.

Contributors guide: ./CONTRIBUTING.md
-->

## Checklist
<!--
Remove items that do not apply. For completed items, change [ ] to [x] or you can click the checkboxes once your 
pull-request is published.
-->

- [ ] Keep pull requests small so they can be easily reviewed.
- [ ] Update the documentation, please follow the [numpy style guide](https://numpydoc.readthedocs.io/en/latest/format.html).
- [ ] Update tests.
- [ ] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `feature`, `ui`, `change`, `documentation`, `breaking`, `ci`
      as they show up in the changelog.
- [ ] Link this PR to related issues by using [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).
- [ ] Add a milestone to the PR for the intended release.
- [ ] Request a review from another developer.

<!--
NOTE: these things are not required to open a PR and can be done afterwards,
while the PR is open.
-->

zoltanszocs committed:
Add title text to table

zoltanszocs committed:
Fix tab key navigation

zoltanszocs committed:
Add missing logic to the remote import

Add checkbox.
Enable/disable checkbox and import button based on selection.
Execute import.

cmutel commented on activity_browser/actions/project/project_remote_import.py:
```diff
@@ -93,27 +103,117 @@ def __init__(self):
+        url_path = self.remote_url_path.text()
+        if url_path[-1] != "/":
+            url_path += "/"
+        self.table.populate(url_path + self.remote_catalogue.text())
```
Let's use `urllib.parse.urljoin` here instead.

zoltanszocs replied:
Done

cmutel reviewed (changes requested):
Nice!
<img width="408" alt="Screenshot 2024-08-12 at 21 05 17" src="https://github.com/user-attachments/assets/b8fe269c-a605-4722-99f9-163e969d6919">
- [ ] Can we add some more space here? It feels quite cramped.
- [ ] Can we please give some indication on *why* the "create project" button is disactivated, i.e. warning text which states that the project name already exists
- [ ] Can we add another checkbox to switch to the project after import? The code to run would be `bw2data.projects.set_current(<new_project_label>)`

zoltanszocs committed:
Added a bit more space between component groups

zoltanszocs committed:
Update dialog flow

Add a message to explain why the import button is disabled.
Disable elements when there is no valid catalogue loaded.
Use urljoin.

zoltanszocs committed:
Add application busy indication
Fix issue with _populate_table calling _check_project_already_exists before the populate call.

zoltanszocs committed:
The overwrite flag wa not reset when changing between two already exisiting projects

zoltanszocs commented:
- Added a bit more space between components - I was not entirely sure where you wanted more space, increasing the global spacing in the layout seemed strange. Let me know how this looks.
- Added status label at the bottom, displaying what is the next necessary step: load catalogue, select project, project already exists
- Added handling for the case when there is no valid catalogue loaded
- Added busy indication for the catalogue loading and project importing
- Added new activate checkbox and functionality
